src/common/components/elements/AppHeader.tsx

```diff
@@ -209,6 +111,124 @@ export function AppHeader() {
   );
 }
 
+function DiscoverMenu(): JSX.Element {
+  const [isTouchAndShowItems, setIsTouchAndShowItems] = useState<boolean>(false);
+  const [isTouch, setIsTouch] = useState<boolean>(false);
+  const [forceHide, setForceHide] = useState<boolean>(false);
+
+  useEffect(() => {
+    setIsTouch(isTouchScreenOnly());
+  }, [setIsTouch]);
+
+  const onClickHeader = useCallback(() => {
+    if (isTouch) setIsTouchAndShowItems(!isTouchAndShowItems);
+  }, [isTouchAndShowItems, isTouch]);
+
+  const onClickHideTemporarily = useCallback(() => {
+    setIsTouchAndShowItems(false);
+    setForceHide(true);
+    setTimeout(() => setForceHide(false), 100);
+  }, []);
+
+  let containerDisplayClasses: string | undefined;
+  let buttonDisplayBasedTextClasses: string;
+  let arrowDownDisplayClasses: string;
+  let arrowUpDisplayClasses: string;
+  let itemContainerDisplayClasses: string;
+  if (!isTouch) {
+    containerDisplayClasses = 'group';
+    buttonDisplayBasedTextClasses = 'text-gray-300 hover:text-white focus:text-white group-hover:text-white group-focus:text-white';
+    arrowDownDisplayClasses = forceHide ? 'block' : 'block group-hover:hidden group-focus:hidden';
+    arrowUpDisplayClasses = forceHide ? 'hidden' : 'hidden group-hover:block group-focus:block';
+    itemContainerDisplayClasses = forceHide ? 'hidden' : 'hidden group-hover:block group-focus:block';
+  } else if (isTouchAndShowItems) {
+    buttonDisplayBasedTextClasses = 'text-white group-focus:text-white';
+    arrowDownDisplayClasses = forceHide ? 'block' : 'hidden group-focus:block';
+    arrowUpDisplayClasses = forceHide ? 'hidden' : 'block group-focus:hidden';
+    itemContainerDisplayClasses = forceHide ? 'hidden' : 'block group-focus:block';
+  } else {
+    buttonDisplayBasedTextClasses = 'text-gray-300 group-focus:text-white';
+    arrowDownDisplayClasses = 'block';
+    arrowUpDisplayClasses = 'hidden';
+    itemContainerDisplayClasses = forceHide ? 'hidden' : 'hidden group-focus:block';
+  }
+
+  return (
+    <div
+      className={classNames('relative inline-block', containerDisplayClasses)}
+      // setTimeout is a hack to allow the click to propagate to the menu item before closing
+      onBlur={() => setTimeout(() => setIsTouchAndShowItems(false), 50)}
```

Comment:
Thanks for the comment but you speak a little more to this what does the menu need? Is this an issue with nested click handlers?